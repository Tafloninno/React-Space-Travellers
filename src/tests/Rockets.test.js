import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import Rockets from '../pages/rockets';

const mockStore = configureMockStore();
const store = mockStore({
  rockets: [
    {
      id: 'falcon1',
      name: 'Falcon 1',
      description:
        'The first privately developed liquid-fueled orbital launch vehicle.',
      flickr_images: 'https://www.example.com/falcon1.jpg',
      reserved: false,
    },
    {
      id: 'falcon9',
      name: 'Falcon 9',
      description:
        'A partially reusable medium-lift launch vehicle designed and manufactured by SpaceX.',
      flickr_images: 'https://www.example.com/falcon9.jpg',
      reserved: true,
    },
  ],
});

describe('Rockets component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Rockets />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
