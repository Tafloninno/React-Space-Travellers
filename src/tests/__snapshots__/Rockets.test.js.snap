// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets component renders correctly 1`] = `
<div
  className="rocket-container"
>
  <div
    className="rocket"
  >
    <img
      alt="falcon1"
      src="https://www.example.com/falcon1.jpg"
    />
    <div
      className="rocket-detail"
    >
      <h2>
        Falcon 1
      </h2>
      <p>
        The first privately developed liquid-fueled orbital launch vehicle.
      </p>
      <button
        onClick={[Function]}
        type="button"
      >
        Reserve Rocket
      </button>
    </div>
  </div>
  <div
    className="rocket"
  >
    <img
      alt="falcon9"
      src="https://www.example.com/falcon9.jpg"
    />
    <div
      className="rocket-detail"
    >
      <h2>
        Falcon 9
      </h2>
      <p>
        <span
          className="reserve"
        >
          Reserved
        </span>
        A partially reusable medium-lift launch vehicle designed and manufactured by SpaceX.
      </p>
      <button
        className="cancel-btn"
        onClick={[Function]}
        type="button"
      >
        Cancel Reservation
      </button>
    </div>
  </div>
</div>
`;
